packages/manager/src/utilities/filterPlanSelectionsByType.ts

```diff
@@ -0,0 +1,47 @@
+import { PlanSelectionType } from 'src/features/linodes/LinodesCreate/SelectPlanPanel';
+
+interface FilteredPlansTypes {
+  nanodes: PlanSelectionType[];
+  standard: PlanSelectionType[];
+  highMem: PlanSelectionType[];
+  proDedicated: PlanSelectionType[];
+  dedicated: PlanSelectionType[];
+  gpu: PlanSelectionType[];
+  metal: PlanSelectionType[];
+  premium: PlanSelectionType[];
+}
+
+export const filterPlanSelectionsByType = (types: PlanSelectionType[]) => {
+  const filteredPlansByType: FilteredPlansTypes = {
+    nanodes: [],
+    standard: [],
+    highMem: [],
+    proDedicated: [],
+    dedicated: [],
+    gpu: [],
+    metal: [],
+    premium: [],
+  };
+
+  types.forEach((type) => {
+    if (/nanode/.test(type.class)) {
+      filteredPlansByType.nanodes.push(type);
+    } else if (/standard/.test(type.class)) {
+      filteredPlansByType.standard.push(type);
+    } else if (/highmem/.test(type.class)) {
+      filteredPlansByType.highMem.push(type);
+    } else if (/prodedicated/.test(type.class)) {
+      filteredPlansByType.proDedicated.push(type);
+    } else if (/^dedicated/.test(type.class)) {
+      filteredPlansByType.dedicated.push(type);
+    } else if (/gpu/.test(type.class)) {
+      filteredPlansByType.gpu.push(type);
+    } else if (/metal/.test(type.class)) {
+      filteredPlansByType.metal.push(type);
+    } else if (/premium/.test(type.class)) {
+      filteredPlansByType.premium.push(type);
+    }
+  });
```

Comment:
What do you think about using a for loop here? I find them more readable.

Also, I think we'd be okay to check for strict equality.

I wonder if a `switch` statement would work nicely here?

```suggestion
  for (const type of types) {
    if (type.class === 'nanode') {
      filteredPlansByType.nanodes.push(type);
    } else if (type.class === 'standard') {
      filteredPlansByType.standard.push(type);
    } else if (type.class === 'highmem') {
      filteredPlansByType.highMem.push(type);
    } else if (type.class === 'prodedicated') {
      filteredPlansByType.proDedicated.push(type);
    } else if (type.class === 'dedicated') {
      filteredPlansByType.dedicated.push(type);
    } else if (type.class === 'gpu') {
      filteredPlansByType.gpu.push(type);
    } else if (type.class === 'metal') {
      filteredPlansByType.metal.push(type);
    } else if (type.class === 'premium') {
      filteredPlansByType.premium.push(type);
    }
  }
```

Reply:
@bnussman-akamai yes we could use for of with switch statement.